src/merge.py

```diff
@@ -13,10 +13,12 @@ def merge(x: list[int], y: list[int]) -> list[int]:
     """
     i, j = 0, 0
     z = []  # a new list to copy elements into
-    # FIXME: fill out the loop so you merge the lists
-    # until one of them is empty
-    while i < len(x) and j < len(y):
-        break  # FIXME: you shouldn't just break here
-    # At least one of the lists is empty now. Copy the
-    # remainder of the other into z.
-    return z
+    while i < len(x) or j < len(y):
+        if i<len(x):
+            z.append(x[0])
+            x.pop(0)
```

Comment:
I see what you are thinking here, and it isn't a bad idea. Get rid of the bits of the input we no longer need. There are two reasons why you probably don't want to do this, though.

First, it can be an unpleasant surprise to call a function and then find out that it has deleted the data you gave it. Sometimes this is necessary, and there is nothing you can do about it (except document this behaviour carefully). But for merging two lists into a third, where it isn't necessary to modify the function arguments, it would likely cause confusion and a substantial amount of swearing. And if history teaches us anything, it is that you are the one doing the swearing when you call the functions many moons after you wrote it.

The second reason you don't know about yet because we haven't talked about computational complexity. Lists are a little weird when it comes to how efficient it is to add and remove elements. The rule of thumb is this: you can add and remove at the right end (with `.append()` and `.pop()`) efficiently, but if you access the other end (with `.insert()` or `.pop(0)`) then you need to copy all the elements you currently have, which for large lists can be quite time-consuming. Because you `x.pop(0)`, you copy the entire `x` list. It changes the algorithm from linear to quadratic in running time (and if that doesn't mean anything to you now, then wait two weeks, and it should).

It is an excellent solution, though, except for those two issues (that you couldn't have known about), so good work!